lwgenerate/azure/azure.go

```diff
@@ -0,0 +1,460 @@
+	AdServicePrincipalId string
+
+	// Should we use the management group, rather than subscription
+	ManagementGroup bool
+
+	// Management Group ID to set
+	ManagementGroupId string
+
+	// List of subscription Ids
+	SubscriptionIds []string
+
+	// Grant read access to ALL subscriptions
+	AllSubscriptions bool
+
+	// Storage Account name
+	StorageAccountName string
+
+	// Storage Account Resource Group
+	StorageAccountResourceGroup string
+
+	// Should we use existing storage account
+	ExistingStorageAccount bool
+
+	// Azure region where the storage account for loggin resides
+	Location string
+}
+
+// Ensure all combinations of inputs are valid for supported spec
+func (args *GenerateAzureTfConfigurationArgs) validate() error {
+	// Validate one of config or activity log was enabled; otherwise error out
+	if !args.ActivityLog && !args.Config {
+		return errors.New("audit log or config integration must be enabled")
+	}
+
+	// Validate that active directory settings are correct
+	if !args.AdIntegration {
+		if args.AdApplicationId == "" || args.AdServicePrincipalId == "" || args.AdApplicationPassword == "" {
+			return errors.New("Active directory details must be set")
+		}
+	}
+
+	// Validate the Mangement Group
+	if args.ManagementGroup && args.ManagementGroupId == "" {
+		return errors.New("When Group Management is enabled, then Group Id must be configured")
+	}
+
+	// Validate Storage Account
+	if args.ExistingStorageAccount {
+		if args.StorageAccountName == "" || args.StorageAccountResourceGroup == "" {
+			return errors.New("When using existing storage account, storage account details must be configured")
+		}
+	}
+
+	return nil
+}
+
+type AzureTerraformModifier func(c *GenerateAzureTfConfigurationArgs)
+
+// NewTerraform returns an instance of the GenerateAzureTfConfigurationArgs struct with the provided enabled
+// settings (config/activity log).
+//
+// Note: Additional configuration details may be set using modifiers of the AzureTerraformModifier type
+//
+func NewTerraform(enableConfig bool, enableActivityLog bool, mods ...AzureTerraformModifier) *GenerateAzureTfConfigurationArgs {
+	config := &GenerateAzureTfConfigurationArgs{ActivityLog: enableActivityLog, Config: enableConfig}
+	for _, m := range mods {
+		m(config)
+	}
+	return config
+}
+
+// WithConfigIntegrationName Set the Config Integration name to be displayed on the Lacework UI
+func WithConfigIntegrationName(name string) AzureTerraformModifier {
+	return func(c *GenerateAzureTfConfigurationArgs) {
+		c.ConfigIntegrationName = name
+	}
+}
+
+// WithAuditLogIntegrationName Set the Config Integration name to be displayed on the Lacework UI
+func WithAuditLogIntegrationName(name string) AzureTerraformModifier {
+	return func(c *GenerateAzureTfConfigurationArgs) {
+		c.ActivityLogIntegrationName = name
+	}
+}
+
+// WithAdIntegration Set the Config Integration name to be displayed on the Lacework UI
+func WithAdIntegration(enableAdIntegration bool) AzureTerraformModifier {
```

Comment:
If I set this to false I get the following error:

```
invalid inputs: Active directory details must be set
```

Perhaps this shouldn't accept a value and just always be true when used?

Reply:
It is the case that the user must supply active directory attributes if this is set to false, these are 

```
	// Active Directory application Id
	AdApplicationId string

	// Active Directory password
	AdApplicationPassword string

	// Active Directory Enterprise app object id
	AdServicePrincipalId string
```
So if these are not set then the user should me made aware of the error

Reply:
OK; so seems like this is an extension of the conversation here: https://github.com/lacework/go-sdk/pull/682#discussion_r805882285

Reply:
I'm somewhat confused by `WithAdIntegration`
Is this intended to cover the `use_existing_ad_application` setting from the Terraform registry?
Or is this simply asking the user if they want to explicitly show the azure ad module in the top-level terraform code?

If the latter, why are we requiring `AdApplicationId`, `AdApplicationPassword`, `AdServicePrincipalId`.
If the former, should we rename this setting to `use_existing_ad_application` instead?

Reply:
I have changed this to be `CreateAdIntegration` as that is what it is, with this set we add in terraform to create an AD Application and we put in the details that are returned from the terraform which is 
`AdApplicationId`, `AdApplicationPassword`, `AdServicePrincipalId` if we do not want to create an AD Application then we need to reference an already existing one and the user must supply these attributes